packages/notification-user-lifecycle-consumer/src/services/messageProcessor.ts

```diff
@@ -0,0 +1,182 @@
+          await interopTokenGenerator.generateInternalToken();
+        await notificationConfigProcessClient.createUserDefaultNotificationConfig(
+          {
+            userId,
+            tenantId,
+          },
+          {
+            headers: {
+              "X-Correlation-Id": correlationId,
+              Authorization: `Bearer ${serialized}`,
+            },
+          }
+        );
+      } catch (err) {
+        loggerInstance.error(
+          `Error creating user default notification config for user ${userId} from tenant ${tenantId}. Reason: ${err}`
+        );
+      }
+      return userServiceSQL.insertUser(userData);
+    })
+    .with("update", async () => {
+      loggerInstance.info(`Update user id ${userId} from tenant ${tenantId}`);
+      return userServiceSQL.updateUser(userData);
+    })
+    .with("delete", async () => {
+      loggerInstance.info(`Removing user ${userId} from tenant ${tenantId}`);
+      try {
+        const { serialized } =
+          await interopTokenGenerator.generateInternalToken();
+        await notificationConfigProcessClient.deleteUserNotificationConfig(
+          undefined,
+          {
+            params: {
+              userId,
+              tenantId,
+            },
+            headers: {
+              "X-Correlation-Id": correlationId,
+              Authorization: `Bearer ${serialized}`,
+            },
+          }
+        );
+      } catch (err) {
+        loggerInstance.error(
+          `Error deleting user default notification config for user ${userId} from tenant ${tenantId}. Reason: ${err}`
+        );
+      }
+      return userServiceSQL.deleteUser(userId);
+    })
+    .exhaustive();
+}
+
+function jsonSafeParse(json: string): unknown {
+  try {
+    return JSON.parse(json);
+  } catch (e) {
+    return {};
+  }
+}
+
+export function messageProcessorBuilder(
+  readModelServiceSQL: ReadModelServiceSQL,
+  userServiceSQL: UserServiceSQL,
+  notificationConfigProcessClient: ReturnType<
+    typeof notificationConfigApi.createProcessApiClient
+  >,
+  interopTokenGenerator: InteropTokenGenerator
+): { processMessage: (payload: EachMessagePayload) => Promise<void> } {
+  return {
+    processMessage: async ({
+      message,
+      partition,
+    }: EachMessagePayload): Promise<void> => {
+      const correlationId: CorrelationId = generateId();
+
+      const loggerInstance = logger({
+        serviceName: "selfcare-client-users-updater",
+        correlationId,
+      });
+
+      try {
+        loggerInstance.info(
+          `Consuming message for partition ${partition} with offset ${message.offset}`
+        );
+
+        const payload = jsonSafeParse(message.value?.toString() ?? "");
+
+        const userEventPayload = UsersEventPayload.safeParse(payload);
+
+        if (!userEventPayload.success) {
+          loggerInstance.warn(
+            `Skipping message for partition ${partition} with offset ${message.offset} - Invalid payload. ${userEventPayload.error}`
```

Comment:
not sure if we want to log the error: it would be useful, but could contain personal data – is this ok since we'll have limited retention?